libs/wingsdk/src/target-sim/function.inflight.ts

```diff
@@ -56,10 +58,35 @@ export class Function implements IFunctionClient, ISimulatorResourceInstance {
 
   public async save(): Promise<void> {}
 
-  public async plan(): Promise<UpdatePlan> {
-    // for now, always replace because we can't determine if the function code
-    // has changed since the last update. see https://github.com/winglang/wing/issues/6116
-    return UpdatePlan.REPLACE;
+  public async plan(invalidated: boolean): Promise<UpdatePlan> {
+    if (invalidated) {
+      return UpdatePlan.REPLACE;
+    }
+
+    await this.createBundlePromise;
+    if (!this.bundle) {
+      throw new Error("Bundle not created");
+    }
```

Comment:
This is a repeating pattern, maybe package up as a private method and reuse across this file?

```js
await this.ensureBundled();
```